cpp/src/arrow/csv/reader.cc

```diff
@@ -970,6 +973,150 @@ class StreamingReaderImpl : public ReaderMixin,
   std::shared_ptr<std::atomic<int64_t>> bytes_decoded_;
 };
 
+/////////////////////////////////////////////////////////////////////////
+// Base class for streaming readers
+
+class TonyReaderImpl : public ReaderMixin<std::shared_ptr<io::RandomAccessFile>>,
```

Comment:
`TonyReader` has a nice ring to it but I think `StreamingFileReader` might be easier to understand for future readers.

Reply:
Ah thanks for pointing this out. I forgot to change this!! :-P